step/step.go

```diff
@@ -153,7 +153,7 @@ func (s SimulatorStarter) prepareSimulator(simulator destination.Device, waitFor
 	}
 
 	UDID := simulator.ID
-	if shouldReset || simulator.Arch != "" {
+	if shouldReset {
```

Comment:
Discussed this with @lpusok, it was probably incorrect and caused by the poor E2E testing that didn't shut down simulators across the tests. Since the Rosetta E2E test came after the basic test case, it detected the first tests's simulator as a "matching" device (despite the arch difference) and needed a reset.